src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -1236,79 +1237,86 @@ internal static SourceHashAlgorithm TryParseHashAlgorithmName(string arg)
-                pattern = pattern.Trim(s_searchPatternTrimChars);
-                bool singleDotPattern = string.Equals(pattern, ".", StringComparison.Ordinal);
-
-                if (!singleDotPattern)
+            if (!singleDotPattern)
+            {
+                while (true)
                 {
-                    while (true)
+                    string? resolvedPath = null;
+                    try
                     {
-                        string? resolvedPath = null;
-                        try
-                        {
-                            if (enumerator == null)
-                            {
-                                enumerator = EnumerateFiles(resolvedDirectoryPath, pattern, searchOption).GetEnumerator();
-                            }
-
-                            if (!enumerator.MoveNext())
-                            {
-                                break;
-                            }
-
-                            resolvedPath = enumerator.Current;
-                        }
-                        catch
+                        if (enumerator == null)
                         {
-                            resolvedPath = null;
+                            enumerator = EnumerateFiles(resolvedDirectoryPath, pattern, searchOption).GetEnumerator();
                         }
 
-                        if (resolvedPath != null)
+                        if (!enumerator.MoveNext())
                         {
-                            // just in case EnumerateFiles returned a relative path
-                            resolvedPath = FileUtilities.ResolveRelativePath(resolvedPath, baseDirectory);
+                            break;
                         }
 
-                        if (resolvedPath == null)
+                        resolvedPath = enumerator.Current;
+                        if (path == "//langversion:?")
                         {
-                            errors.Add(Diagnostic.Create(MessageProvider, (int)MessageProvider.FTL_InvalidInputFileName, path));
-                            break;
+                            // Ugly check to see what's going on CI :(
+                            var sb = new StringBuilder();
+                            sb.Append($"directory: '{directory}', ");
+                            sb.Append($"baseDirectory: '{baseDirectory}', ");
+                            sb.Append($"resolvedDirectoryPath: '{resolvedDirectoryPath}', ");
+                            var enumeratedFiles = EnumerateFiles(resolvedDirectoryPath, pattern, searchOption).ToArray();
+                            sb.Append($"Enumerating {enumeratedFiles.Length} files: ");
+                            foreach (var enumeratedFile in enumeratedFiles)
+                            {
+                                sb.Append($"'{enumeratedFile}', ");
+                            }
+                            throw new Exception(sb.ToString());
                         }
```

Comment:
I know this is a too bad ugly approach for investigating the issue :smile:
But anyway, at this point, there are the following variables and their values:

|Variable|Value|
|--------|------|
|`directory`|`"/"`|
|`baseDirectory`|`"/tmp/RoslynTests"`|
|`resolvedDirectoryPath`|`"/"`|

- I'm not sure if `resolvedDirectoryPath` is supposed to be "/" or not. This comes from `FileUtilities.ResolveRelativePath(directory, baseDirectory)`.
- There is an empty file named "1" in "/" directory. I don't think this is something expected. I don't know what's writing this file.

cc @jaredpar